### PR TITLE
block player one from selecting a block with the wrong value

### PR DESCRIPTION
**Pull request by kwiedenhaupt**
used currentlySelectedBlock to store the index of the previously clicked block in order to be used later in the code.

### CODE REVIEW

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:79]: don't merge commented out code

File: redux/longestStreakSlice.ts
Comment (vithushan19) [inline redux/longestStreakSlice.ts:119]: don't merge this in. please remove

**Review by vithushan19 — COMMENTED** [review]
You have too much commented out code, but everything else looks great. Please get Abhi to review your code and you can merge it once you get his approval

**Comment by vithushan19** [comment]
looks like your tests are also failing and need to be fixed

**Comment by vithushan19** [comment]
Your build is failing. Try `npm run build` to see the errors.

**Comment by vithushan19** [comment]
Let me know when your build is passing all the checks and then I'll review. You might want to ask abhi to review before me.

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:78]: Do you understand what's happening here? This is called json destructing

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:143]: ... ?

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:22]: Why is this needed?

File: redux/longestStreakSlice.ts
Comment (vithushan19) [inline redux/longestStreakSlice.ts:65]: Why is this property needed?

Comment (vithushan19) [reply redux/longestStreakSlice.ts:65]: When is it used? When is it set to true?

File: redux/longestStreakSlice.ts
Comment (vithushan19) [inline redux/longestStreakSlice.ts:132]: Cool!

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (vithushan19) [reply pages/studentPortal/labs/multiplication/game.tsx:22]: If it's for ai selection why did you make this property an optional?

**Review by vithushan19 — APPROVED** [review]
Nice! I had a few questions but it looks good to me.

I think we are almost done with this game. Next week you will learn about databases and how to read and write to a db. You will add a concept of a user level for this game. When they beat the first level they progress to their 3 times tables.

When your game is complete we will do a test session with the other students where they will look for bugs in your game. So make sure you are testing different edge cases and you're ready for others to try to break your game.

Great work!

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (kwiedenhaupt) [reply pages/studentPortal/labs/multiplication/game.tsx:78]: I understand that it's tying the component to the state, doing the getting and the setting and tying it to the payload action.  That's just a more succinct way to do it?

File: pages/studentPortal/labs/multiplication/game.tsx
Comment (kwiedenhaupt) [reply pages/studentPortal/labs/multiplication/game.tsx:143]: Screen Shot 2022-09-16 at 7 38 03 PM
adds a fade to transparent gradient to the center of the game board

File: redux/longestStreakSlice.ts
Comment (kwiedenhaupt) [reply redux/longestStreakSlice.ts:65]: It allows me to re-initialize the game board with redux when I hit the reset button on the play_game stage. It gives me new numbers, etc.

Comment (kwiedenhaupt) [reply redux/longestStreakSlice.ts:65]: I had to clear state for the game in order to truly reset. Otherwise, if I just tried to initialize it would remember some parts of the state.

**Pull request merged**
